### DISCUSSION TITLE
Text Input Podcast generation

### DISCUSSION THREAD

**Discussion by prerana016** [souzatharsis/podcastfy#220] [Q&A]
Is there any minimum or maximum limit on input text for generating podcast? (how long short or long the text input can be ?)
How is the duration of podcast generated ?

**Comment by Cfomodz** [comment]
# Input Text Limits

**No strict minimum or maximum limits.** Podcastfy processes any text length, but:
- **Recommended minimum:** 200-300 words for meaningful conversation
- Very long inputs are automatically chunked in longform mode

## Podcast Duration

Duration depends on the mode you choose:

### Shortform (Default)
- **Duration:** 2-5 minutes (fixed)
- Content is summarized to fit this range

```python
generate_podcast(urls=["https://example.com/article"])
```

### Longform
- **Duration:** 20-30+ minutes (adjustable)
- Controlled by two parameters:
  - `max_num_chunks` (default: 8) - more chunks = longer podcast
  - `min_chunk_size` (default: 600 characters) - smaller = more chunks

```python
generate_podcast(
    urls=["https://example.com/article"],
    longform=True
)
```

**To customize duration:**
```python
custom_config = {
    "max_num_chunks": 12,  # Increase for longer
    "min_chunk_size": 400   # Decrease for longer
}

generate_podcast(
    urls=["https://example.com/article"],
    longform=True,
    conversation_config=custom_config
)
```
